### PR TITLE
e2e: Ensure the underlying Bundle unpack Pod is re-created when deleted

### PR DESCRIPTION
**Pull request by timflannagan**
The k8s provisioner currently maintains a Bundle unpack Pod, and parses the Pod's logs in order to read unpacked Bundle contents so it can store it in a series of ConfigMaps. We should ensure we a simple e2e test to ensure this behavior so we're not regressing silently.

### CODE REVIEW

**Comment by timflannagan** [comment]
Wait until #85 lands.

/hold

File: test/e2e/k8s_provisioner_test.go
Comment (timflannagan) [inline test/e2e/k8s_provisioner_test.go:83]: Note: we can extend this logic to also check that the object and metadata ConfigMaps for this Bundle object get re-created. This can be done as a follow-up?

File: test/e2e/k8s_provisioner_test.go
Comment (timflannagan) [reply test/e2e/k8s_provisioner_test.go:105]: Yeah honestly this was my first iteration. I'm not entirely sure why I settled on using a podNN vs. a pod variable but it was also 3am in the morning, so it's possible my brain wasn't working at 100% capacity. It seems like this would allow us to eliminate a single call, so it seems worthwhile to change.

Comment (timflannagan) [reply test/e2e/k8s_provisioner_test.go]: Updated - note: I had to wrap the delete call in an eventually handler as to avoid 1/3 runs being flaky. This seems like a reasonable workaround to me?

File: test/e2e/k8s_provisioner_test.go
Comment (timflannagan) [reply test/e2e/k8s_provisioner_test.go:89]: Added a TODO comment so we don't lose track. It would be nice to differentiate between FIXME's and TODO's, where the latter has a github action that automatically creates follow-up tickets, and the former is a considered a blocker for merging the PR and mainly used in the WIP/draft PR contest. cc @joelanford in case this logic makes sense to you?

File: test/e2e/k8s_provisioner_test.go
Comment (joelanford) [inline test/e2e/k8s_provisioner_test.go:118]: Nit: do we need `Eventually` here?

Comment (joelanford) [reply test/e2e/k8s_provisioner_test.go:118]: Ah, just saw your other comment. That's... interesting. Any clue why its flaky without the eventually?

Comment (timflannagan) [reply test/e2e/k8s_provisioner_test.go:118]: I'm not entirely sure - this is what I had logged in a markdown file when I was debugging yesterday:

```
    Expected
        <*errors.StatusError | 0xc0003c0aa0>: {
            ErrStatus: {
                TypeMeta: {Kind: "", APIVersion: ""},
                ListMeta: {
                    SelfLink: "",
                    ResourceVersion: "",
                    Continue: "",
                    RemainingItemCount: nil,
                },
                Status: "Failure",
                Message: "pods \"plain-unpack-bundle-olm-crds-valid\" not found",
                Reason: "NotFound",
                Details: {
                    Name: "plain-unpack-bundle-olm-crds-valid",
                    Group: "",
                    Kind: "pods",
                    UID: "",
                    Causes: nil,
                    RetryAfterSeconds: 0,
                },
                Code: 404,
            },
        }
    to be nil
```

My initial thought was an object cache issue but we create the c-r client.Client using client.New so we're likely directly reading/writing to the apiserver throughout the e2e suite. Maybe the `CreateOrRecreate(...)` function is deleting the Pod when it detects we need to make changes to the underlying bundle unpack pod?

I realize I could also just wrap the Delete call with `client.IgnoreNotFound(c.Delete(ctx, pod))` to burn down the flakes, which seems fine to me, but I'm not sure if there's the possibility we sweep legitimate controller issues under the rug over time.

**Review by exdx — APPROVED** [review]

File: test/e2e/k8s_provisioner_test.go
Comment (joelanford) [reply test/e2e/k8s_provisioner_test.go]: Somewhat mystifying that it exists for the `List` but not the `Get`. If it was a caching issue, `Get` would always have it if a previous `List` did (assuming it didn't get deleted in-between).

`CreateOrRecreate()` is an interesting thought.

> there's the possibility we sweep legitimate controller issues under the rug over time.
I think this is deserving of a little more attention for this reason.

Comment (timflannagan) [reply test/e2e/k8s_provisioner_test.go]: Added a TODO comment for re-evaluating this in #105. Any existing TODO comments should be created as issues in #106.

Comment (joelanford) [reply test/e2e/k8s_provisioner_test.go]: I think I figured it out: https://github.com/operator-framework/rukpak/pull/107

**Pull request merged**
